### PR TITLE
Sysctl: validate that setting a pod sysctl does not affect node and other pods

### PR DESCRIPTION
**Pull request by mmirecki**

### CODE REVIEW

**Comment by deads2k** [comment]
/approve

this seems reasonable to test, but I don't speak the details of sysctls, so that will need detailed review
/assign @mrunalp

**Comment by mmirecki** [comment]
/cc @danwinship 
Could you please review? Acked by deads2k already

File: test/extended/security/sysctl.go [hunk 1]
```diff
@@ -34,4 +35,99 @@ var _ = g.Describe("[sig-arch] [Conformance] sysctl", func() {
...
+		f := oc.KubeFramework()
+
+		nodePod := frameworkpod.CreateExecPodOrFail(f.ClientSet, f.Namespace.Name, "sysctl-pod-", func(pod *v1.Pod) {
+			pod.Spec.Volumes = []v1.Volume{
+				{Name: "sysvolume", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/proc"}}},
+			}
+			pod.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{{Name: "sysvolume", MountPath: "/host/proc"}}
+		})
+
+		nodeOutputBeforeSysctlAplied, err := oc.AsAdmin().Run("exec").Args(nodePod.Name, "--", "/bin/bash", "-c", "cat /host/"+path).Output()
```
Comment (danwinship) [inline test/extended/security/sysctl.go:49]:
```suggestion
		nodeOutputBeforeSysctlAplied, err := oc.AsAdmin().Run("exec").Args(nodePod.Name, "--", "cat", "/host/"+path).Output()
```
? (and likewise elsewhere)

**Review by danwinship — COMMENTED** [review]
These aren't really distinct tests; they're more like pre/post-conditions on the existing "pod should start for each sysctl on whitelist" test. It seems like it would be better to use a `BeforeEach` to create a pre-existing pod and dump the node sysctl state, and an `AfterEach` to re-check the pre-existing pod, re-check the node state, and test a newly-created pod. (This would also result in creating a lot fewer pods total, speeding the test up.)

**Comment by mmirecki** [comment]
> These aren't really distinct tests; they're more like pre/post-conditions on the existing "pod should start for each sysctl on whitelist" test. It seems like it would be better to use a `BeforeEach` to create a pre-existing pod and dump the node sysctl state, and an `AfterEach` to re-check the pre-existing pod, re-check the node state, and test a newly-created pod. (This would also result in creating a lot fewer pods total, speeding the test up.)

Modified to check everything in one test. Will not use `BeforeEach` as the pre-existing items need access to table paramters. Instead grouped the various test check in separate By's

**Comment by danwinship** [comment]
(nitpicks but this one looks basically ready)

File: test/extended/security/sysctl.go [hunk 2]
```diff
@@ -11,27 +11,67 @@ import (
...
+			o.Expect(podOutput).Should(o.Equal(defaultSysctlValue))
+		})
 	},
-		t.Entry("kernel.shm_rmid_forced", "kernel.shm_rmid_forced", "1", "/proc/sys/kernel/shm_rmid_forced"),
-		t.Entry("net.ipv4.ip_local_port_range", "net.ipv4.ip_local_port_range", "32768\t61000", "/proc/sys/net/ipv4/ip_local_port_range"),
-		t.Entry("net.ipv4.tcp_syncookies", "net.ipv4.tcp_syncookies", "1", "/proc/sys/net/ipv4/tcp_syncookies"),
-		t.Entry("net.ipv4.ping_group_range", "net.ipv4.ping_group_range", "1\t0", "/proc/sys/net/ipv4/ping_group_range"),
-		t.Entry("net.ipv4.ip_unprivileged_port_start", "net.ipv4.ip_unprivileged_port_start", "1002", "/proc/sys/net/ipv4/ip_unprivileged_port_start"),
+		t.Entry("kernel.shm_rmid_forced", "kernel.shm_rmid_forced", "1", "/proc/sys/kernel/shm_rmid_forced", "0"),
+		t.Entry("net.ipv4.ip_local_port_range", "net.ipv4.ip_local_port_range", "32769\t61001", "/proc/sys/net/ipv4/ip_local_port_range", "32768\t60999"),
```
[the remark this answers is not in the document]
Comment (mmirecki) [reply test/extended/security/sysctl.go:72]: I condensed multiple table tests into 1, taking the values from the last table. No previous issues with any of the values.
The new values for the pod to start ("value") should actually be better, as they are not the "default".

File: test/extended/security/sysctl.go [hunk 3]
```diff
@@ -11,27 +11,67 @@ import (
...
-		o.Expect(output).Should(o.Equal(value))
+		g.By("create a preexisting pod to validate sysctl are not applied on it", func() {
+			preexistingPod = frameworkpod.CreateExecPodOrFail(f.ClientSet, f.Namespace.Name, "sysctl-pod-", func(pod *v1.Pod) {
+				pod.Spec.NodeName = nodePod.Spec.NodeName
+			})
+			previousPodSysctlValue, err = oc.AsAdmin().Run("exec").Args(preexistingPod.Name, "--", "cat", path).Output()
+			o.Expect(err).NotTo(o.HaveOccurred(), "unable to check sysctl value")
+		})
 
+		g.By("creates a pod with a sysctl", func() {
```
[the remark this answers is not in the document]
Comment (mmirecki) [reply test/extended/security/sysctl.go:39]: Fixed to use gerunds.

File: test/extended/security/sysctl.go [hunk 4]
```diff
@@ -11,27 +11,67 @@ import (
...
+			})
+			nodeOutputBeforeSysctlAplied, err = oc.AsAdmin().Run("exec").Args(nodePod.Name, "--", "cat", "/host/"+path).Output()
+			o.Expect(err).NotTo(o.HaveOccurred(), "unable to check sysctl value")
 		})
 
-		g.By("checking that the sysctl was set")
-		output, err := oc.AsAdmin().Run("exec").Args(tuningTestPod.Name, "--", "/bin/bash", "-c", "cat "+path).Output()
-		o.Expect(err).NotTo(o.HaveOccurred(), "unable to check sysctl value")
-		o.Expect(output).Should(o.Equal(value))
+		g.By("create a preexisting pod to validate sysctl are not applied on it", func() {
```
[the remark this answers is not in the document]
Comment (mmirecki) [reply test/extended/security/sysctl.go]: done

**Pull request merged**